core/corehttp/gateway_test.go

```diff
@@ -129,7 +129,6 @@ func newTestServerAndNode(t *testing.T, ns mockNamesys) (*httptest.Server, iface
 
 	dh.Handler, err = makeHandler(n,
 		ts.Listener,
-		HostnameOption(),
```

Comment:
:thought_balloon: @hacdias was too late for me to figure out why `TestGatewayGet` in https://app.circleci.com/pipelines/github/ipfs/kubo/9010/workflows/15e0dfe2-0ee6-4e86-8056-c9cb7d65521c/jobs/97462/steps?invite=true#step-104-2633 fails, but we should not ignore it, every test is there for a reason.

`broken.example.com` seems to tests two distinct things:
1. what happens when DNSLink or IPNS points at invalid content path (here, it resolved to `/ipns/{k.Cid()}` where `k` is not a libp2p-key, but a regular CID, so it should error)
2. what happens when DNSLink gateway (`Host: broken.example.com`)  gets a request for Path gateway (`/ipfs/cid`). It should error, because `/ipns/broken.example.com/ipfs/cid` does not exist.

Perhaps decoupling these and testing each case separately with different domain name will help to unblock?

Reply:
Quick updates regarding my debugging. I found the source of the problem but I'm not yet sure how to proceed because I  don't necessarily agree with what's been done in Kubo right now:

```go
isDNSLinkName(r.Context(), api, "broken.example.com")
```

- `master`: returns `true` because we call `namesys.Resolve` directly with 1 level of recursion at most. And `err == namesys.ErrResolveRecursion` yields true.
- `hostname-libipfs`: returns `false` because we call `ResolvePath` which a) does not have a limit of 1 level of recursion, and b) fails as the name cannot be resolved [but not because of recursion]. So returns `false`. 

By returning `false`, we do not do the rewriting. I feel like the recursion check in Kubo was more of a "patch" that happened to work for this case, but not necessarily the best way of doing this.

Reply:
Thank you for looking  into this, the approach is sensible, but I agree it adds complexity.
If we do this, let's make the most out of it. 

I propose we replace `HasDNSLinkRecord` with  `GetIPNSRecord` to  match what we already do for IPNS records,  and have ability to read the value.